app/src/main/java/space/taran/arknavigator/mvp/presenter/GalleryPresenter.kt

```diff
@@ -40,7 +40,7 @@ class GalleryPresenter(
         viewState.init(previews)
     }
 
-    fun deleteResource(resource: ResourceId) {
+    fun deleteResource(resource: ResourceId) = presenterScope.launch {
```

Comment:
Regarding potential cancellation of coroutines and ways to solve it:
I think `NonCancellable` is essential here, since it guarantees that the coroutine will not be cancelled at all.
All operations writing to tags storage (filesystem) must be not possible to cancel.

Operations writing to index (Room) are less important, because right now it is just cache.
But when we implement more advanced features like version tracking, the index will become more important.
We can make these operations `NonCancellable` as well.